src/utils.ts

```diff
@@ -1,20 +1,34 @@
-import * as https from 'https';
+import axiosBase, { AxiosRequestConfig } from "axios";
+const ProxyAgent = require("proxy-agent");
 
-export function httpsRequest(url: string, options: https.RequestOptions = {}, encoding = 'utf8'): Promise<string> {
-	return new Promise((resolve, reject) => {
-		https.request(url, options, res => {
-			if (res.statusCode === 301 || res.statusCode === 302) { // follow redirects
-				return resolve(httpsRequest(res.headers.location, options, encoding));
-			}
-			if (res.statusCode !== 200) {
-				return reject(res.statusMessage);
-			}
-			let body = '';
-			res.setEncoding(encoding)
-				.on('data', data => body += data)
-				.on('end', () => resolve(body));
-		})
-			.on('error', reject)
-			.end();
-	});
+const httpProxy = process.env["HTTP_PROXY"] || process.env["http_proxy"];
+const httpsProxy = process.env["HTTPS_PROXY"] || process.env["https_proxy"];
+
+let proxyConf = {};
+
+if (httpProxy || httpsProxy) {
+  console.log(`Found proxy setting. Using to download files.`);
+  console.log(`HTTP: ${httpProxy}`);
+  console.log(`HTTPS: ${httpProxy}`);
+  proxyConf = {
+    proxy: false,
+    httpAgent: httpProxy ? new ProxyAgent(httpProxy) : undefined,
+    httpsAgent: httpsProxy ? new ProxyAgent(httpsProxy) : undefined,
+  };
+}
+
+const axios = axiosBase.create({ ...proxyConf });
+
+export async function request(
+  url: string,
+  options: AxiosRequestConfig = {}
+): Promise<any> {
+  console.log(`Request to: ${url} with options: ${JSON.stringify(options)}`);
```

Comment:
```suggestion
```
We shouldn't be printing to the console proxy details

Reply:
What would be the best solution to add debug functionality to check if the correct proxy is used?
The default set up job (on Github Enterprise) also prints proxy configs if setup:
![image](https://user-images.githubusercontent.com/9947112/142840735-412740f5-e2be-4dae-9b91-82391eaed93a.png)

Reply:
This is a good question, and is something that has some variance in implementations. In this case we expect the application using this library to output information like this, not the library itself.

Logging information is an important task of any library, but there isn't a way in this library to currently log anything. A future PR could introduce an interface and method overload that allows logging to different targets, but this is out of scope here.

Reply:
Removed